test/jdk/jdk/incubator/vector/templates/Unit-Miscellaneous.template

```diff
@@ -411,55 +411,89 @@
         }
     }
 
+    static int maskTrueCount(boolean[] a, int idx) {
+        int trueCount = 0;
+        for (int i = idx; i < idx + SPECIES.length(); i++) {
+            trueCount += a[i] ? 1 : 0;
+        }
+        return trueCount;
+    }
+
     @Test(dataProvider = "maskProvider")
     static void maskTrueCount$vectorteststype$SmokeTest(IntFunction<boolean[]> fa) {
         boolean[] a = fa.apply(SPECIES.length());
+#if[Int]
```

Comment:
We already know the length of `r`, it's `a.length`, so we can just do:
```java 
        int[] r = new int[a.length];
```
Thereby reducing the complexity. I realize you are copying the pattern of `lfr` which i think is also unnecessary. Let's clean this up later.

Reply:
Resolved. Thanks for the suggestion!